src/Containers/FormContainer.tsx

```diff
@@ -5,19 +5,22 @@ import { colorTheme } from "src/Themes/Colors";
 import Input from "src/Components/Input";
 import { GradientButton } from "src/Components/Buttons";
 
-type FieldsState = { [key: string]: { value?: string; isValid: boolean } };
+type FieldValue = { value?: string; isValid: boolean };
+type PreValidateFieldValue = { value?: string; isValid: undefined };
```

Comment:
I was reading first  lines of this PR's changes for 3 minutes trying to understand why do we have a separation between `FieldValue` and `PreValidateFieldValue`.

Do we really need this separation? What is the need?

`isValid` is a boolean indicating if the value is valid or not. If the field was not validated yet, it usually is `false` by default. 

What business-logic usecase can be implemented with isValid being `undefined`?

If we will make it `false` by default, the code will become much cleaner and no questions will appear by reading it.